plugins/notifications-backend/src/service/router.ts

```diff
@@ -250,27 +258,75 @@ export async function createRouter(
     res.send(status);
   });
 
-  router.post('/update', async (req, res) => {
+  const updateNotifications = async (
+    user: string,
+    ids: string[],
+    read?: boolean,
+    saved?: boolean,
+  ) => {
+    if (read === true) {
+      await store.markRead({ user, ids });
+    } else if (read === false) {
+      await store.markUnread({ user, ids });
+    }
+
+    if (saved === true) {
+      await store.markSaved({ user: user, ids });
+    } else if (saved === false) {
+      await store.markUnsaved({ user: user, ids });
+    }
+  };
+
+  router.put('/notification/:id', async (req, res) => {
+    const user = await getUser(req);
+    const { read, saved } = req.body;
+    await updateNotifications(user, [req.params.id], read, saved);
+    if (signals) {
+      if (read === true) {
+        await signals.publish<NotificationReadSignal>({
+          recipients: { type: 'user', entityRef: [user] },
+          message: {
+            action: 'notification_read',
+            notification_ids: [req.params.id],
+          },
+          channel: 'notifications',
+        });
+      } else if (read === false) {
+        await signals.publish<NotificationReadSignal>({
+          recipients: { type: 'user', entityRef: [user] },
+          message: {
+            action: 'notification_unread',
+            notification_ids: [req.params.id],
+          },
+          channel: 'notifications',
+        });
+      }
+    }
+
+    const notification = await getNotification(user, req.params.id);
+    if (notification === null) {
+      res.status(404).send({ error: 'Not found' });
+      return;
+    }
+    res.send(notification);
+  });
+
+  router.put('/notifications', async (req, res) => {
```

Comment:
I tend to keep the usage of `PUT` quite strict, as in specifically when you can more or less update an object at a URL and then read the same thing back with a `GET`. If it does any kind of merging I'd use `PATCH`, but in this case for multiple updates I might go with something like `POST /notifications/update` like we had tbh

Reply:
Ok, will change